### PR TITLE
tests/bench*: convert to ztimer

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR converts some `tests/bench_*` to ztimer.

### CODE REVIEW

**Comment by bergzand** [comment]
This PR breaks the printing of the "ticks" result in the output.


<details><summary>With Ztimer on nrf52840dk `tests/bench_mutex_pingpong/`</summary>

```
2021-11-02 10:59:37,741 # Help: Press s to start test, r to print it is ready
s
2021-11-02 10:59:39,327 # START
2021-11-02 10:59:39,332 # main(): This is RIOT! (Version: 2022.01-devel-326-gee721-HEAD)
2021-11-02 10:59:39,334 # main starting
2021-11-02 10:59:40,325 # { "result" : 115108 }
```

</details>

<details><summary>current master on nrf52840dk `tests/bench_mutex_pingpong/`</summary>

```
2021-11-02 11:00:16,351 # Help: Press s to start test, r to print it is ready
s
2021-11-02 11:00:17,230 # START
2021-11-02 11:00:17,236 # main(): This is RIOT! (Version: 2022.01-devel-318-gf64b16-HEAD)
2021-11-02 11:00:17,237 # main starting
2021-11-02 11:00:18,229 # { "result" : 115108, "ticks" : 555 }
```

</details>

**Comment by kaspar030** [comment]
> This PR breaks the printing of the "ticks" result in the output.

thanks for looking at the actual output. the benchmarks were missing an include for "periph_conf.h" and "timex.h", which previously were included through xtimer.h.

**Comment by bergzand** [comment]
Both on the nrf52840dk

<details> <summary> `tests/bench_sys_atomic_utils` on master </summary>

```
2021-11-03 10:20:57,349 # +------+----------+------+------------------+------------------+------------------+
2021-11-03 10:20:57,356 # | mode |       op | bits |         volatile |      atomic util |       c11 atomic |
2021-11-03 10:20:57,363 # +------+----------+------+------------------+------------------+------------------+
2021-11-03 10:20:57,400 # | atom |    store |    8 |          6252 µs |         12501 µs |         10939 µs |
2021-11-03 10:20:57,437 # | atom |    store |   16 |          6252 µs |         12501 µs |         10939 µs |
2021-11-03 10:20:57,470 # | atom |    store |   32 |          6252 µs |          9377 µs |          9377 µs |
2021-11-03 10:20:57,549 # | atom |    store |   64 |          9376 µs |         17189 µs |         45314 µs |
2021-11-03 10:20:57,586 # | atom |     load |    8 |          7815 µs |         10939 µs |         10939 µs |
2021-11-03 10:20:57,623 # | atom |     load |   16 |          7814 µs |         10939 µs |         10939 µs |
2021-11-03 10:20:57,660 # | atom |     load |   32 |          7815 µs |         10939 µs |         10939 µs |
2021-11-03 10:20:57,723 # | atom |     load |   64 |          9376 µs |         15627 µs |         31252 µs |
2021-11-03 10:20:57,785 # | atom |      add |    8 |         12501 µs |         20314 µs |         21877 µs |
2021-11-03 10:20:57,850 # | atom |      add |   16 |         14064 µs |         18751 µs |         25002 µs |
2021-11-03 10:20:57,910 # | atom |      add |   32 |         10939 µs |         17189 µs |         25002 µs |
2021-11-03 10:20:58,017 # | atom |      add |   64 |         17189 µs |         25002 µs |         57814 µs |
2021-11-03 10:20:58,075 # | atom |      sub |    8 |         12502 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:58,134 # | atom |      sub |   16 |         12502 µs |         18752 µs |         20314 µs |
2021-11-03 10:20:58,200 # | atom |      sub |   32 |         10939 µs |         23439 µs |         25001 µs |
2021-11-03 10:20:58,318 # | atom |      sub |   64 |         18752 µs |         23439 µs |         68752 µs |
2021-11-03 10:20:58,383 # | atom |       or |    8 |         15626 µs |         21877 µs |         20315 µs |
2021-11-03 10:20:58,448 # | atom |       or |   16 |         18752 µs |         18752 µs |         20314 µs |
2021-11-03 10:20:58,505 # | atom |       or |   32 |         10940 µs |         18752 µs |         20314 µs |
2021-11-03 10:20:58,624 # | atom |       or |   64 |         17189 µs |         35939 µs |         59377 µs |
2021-11-03 10:20:58,683 # | atom |      xor |    8 |         12501 µs |         18752 µs |         20315 µs |
2021-11-03 10:20:58,752 # | atom |      xor |   16 |         17190 µs |         25001 µs |         20314 µs |
2021-11-03 10:20:58,817 # | atom |      xor |   32 |         10939 µs |         21876 µs |         25002 µs |
2021-11-03 10:20:58,945 # | atom |      xor |   64 |         25002 µs |         31251 µs |         65627 µs |
2021-11-03 10:20:59,004 # | atom |      and |    8 |         12502 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:59,061 # | atom |      and |   16 |         10939 µs |         18752 µs |         20315 µs |
2021-11-03 10:20:59,126 # | atom |      and |   32 |         10939 µs |         21877 µs |         25002 µs |
2021-11-03 10:20:59,251 # | atom |      and |   64 |         25001 µs |         32814 µs |         60939 µs |
2021-11-03 10:20:59,308 # | atom |      set |    8 |         12502 µs |         17190 µs |         20315 µs |
2021-11-03 10:20:59,376 # | atom |      set |   16 |         14064 µs |         20314 µs |         26564 µs |
2021-11-03 10:20:59,432 # | atom |      set |   32 |         10939 µs |         17189 µs |         20314 µs |
2021-11-03 10:20:59,571 # | atom |      set |   64 |         26565 µs |         32814 µs |         73439 µs |
2021-11-03 10:20:59,628 # | atom |    clear |    8 |         10939 µs |         18752 µs |         20314 µs |
2021-11-03 10:20:59,691 # | atom |    clear |   16 |         14064 µs |         21877 µs |         20314 µs |
2021-11-03 10:20:59,759 # | atom |    clear |   32 |         15627 µs |         18752 µs |         26564 µs |
2021-11-03 10:20:59,888 # | atom |    clear |   64 |         25001 µs |         31252 µs |         65627 µs |
2021-11-03 10:20:59,953 # | semi |      add |    8 |         14065 µs |         18752 µs |         25001 µs |
2021-11-03 10:21:00,024 # | semi |      add |   16 |         14064 µs |         25001 µs |         25002 µs |
2021-11-03 10:21:00,089 # | semi |      add |   32 |         10940 µs |         21877 µs |         25002 µs |
2021-11-03 10:21:00,211 # | semi |      add |   64 |         25001 µs |         32815 µs |         57815 µs |
2021-11-03 10:21:00,283 # | semi |      sub |    8 |         18751 µs |         21877 µs |         25002 µs |
2021-11-03 10:21:00,354 # | semi |      sub |   16 |         14064 µs |         25002 µs |         25002 µs |
2021-11-03 10:21:00,419 # | semi |      sub |   32 |         10939 µs |         21877 µs |         25001 µs |
2021-11-03 10:21:00,536 # | semi |      sub |   64 |         18752 µs |         23439 µs |         68752 µs |
2021-11-03 10:21:00,601 # | semi |       or |    8 |         12502 µs |         25001 µs |         20314 µs |
2021-11-03 10:21:00,666 # | semi |       or |   16 |         18751 µs |         18752 µs |         20315 µs |
2021-11-03 10:21:00,734 # | semi |       or |   32 |         15627 µs |         18752 µs |         26564 µs |
2021-11-03 10:21:00,858 # | semi |       or |   64 |         25002 µs |         32814 µs |         59376 µs |
2021-11-03 10:21:00,918 # | semi |      xor |    8 |         15627 µs |         17190 µs |         20314 µs |
2021-11-03 10:21:00,983 # | semi |      xor |   16 |         12502 µs |         20314 µs |         25001 µs |
2021-11-03 10:21:01,040 # | semi |      xor |   32 |         10939 µs |         18752 µs |         20314 µs |
2021-11-03 10:21:01,168 # | semi |      xor |   64 |         25001 µs |         35939 µs |         60939 µs |
2021-11-03 10:21:01,233 # | semi |      and |    8 |         12502 µs |         20314 µs |         25002 µs |
2021-11-03 10:21:01,298 # | semi |      and |   16 |         12502 µs |         25002 µs |         20314 µs |
2021-11-03 10:21:01,361 # | semi |      and |   32 |         10940 µs |         18752 µs |         26565 µs |
2021-11-03 10:21:01,492 # | semi |      and |   64 |         28127 µs |         35939 µs |         60939 µs |
2021-11-03 10:21:01,500 # +------+----------+------+------------------+------------------+------------------+
```

</details>
<details> <summary> `tests/bench_sys_atomic_utils` this PR </summary>

```
2021-11-03 10:20:05,454 # +------+----------+------+------------------+------------------+------------------+
2021-11-03 10:20:05,461 # | mode |       op | bits |         volatile |      atomic util |       c11 atomic |
2021-11-03 10:20:05,469 # +------+----------+------+------------------+------------------+------------------+
2021-11-03 10:20:05,515 # | atom |    store |    8 |          6251 µs |         23439 µs |          9376 µs |
2021-11-03 10:20:05,561 # | atom |    store |   16 |          6251 µs |         23438 µs |          9376 µs |
2021-11-03 10:20:05,604 # | atom |    store |   32 |          6252 µs |         18751 µs |         10938 µs |
2021-11-03 10:20:05,685 # | atom |    store |   64 |          9376 µs |         17189 µs |         46876 µs |
2021-11-03 10:20:05,736 # | atom |     load |    8 |          7814 µs |         25001 µs |         10939 µs |
2021-11-03 10:20:05,785 # | atom |     load |   16 |         10939 µs |         20313 µs |         10939 µs |
2021-11-03 10:20:05,842 # | atom |     load |   32 |         10939 µs |         26564 µs |         12501 µs |
2021-11-03 10:20:05,906 # | atom |     load |   64 |          9376 µs |         15627 µs |         31251 µs |
2021-11-03 10:20:05,971 # | atom |      add |    8 |         17189 µs |         18751 µs |         21876 µs |
2021-11-03 10:20:06,042 # | atom |      add |   16 |         12501 µs |         23438 µs |         28126 µs |
2021-11-03 10:20:06,103 # | atom |      add |   32 |         10938 µs |         18751 µs |         25001 µs |
2021-11-03 10:20:06,210 # | atom |      add |   64 |         17189 µs |         23439 µs |         59376 µs |
2021-11-03 10:20:06,281 # | atom |      sub |    8 |         12501 µs |         23438 µs |         28126 µs |
2021-11-03 10:20:06,341 # | atom |      sub |   16 |         12501 µs |         18751 µs |         21876 µs |
2021-11-03 10:20:06,403 # | atom |      sub |   32 |         15626 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:06,510 # | atom |      sub |   64 |         17188 µs |         23438 µs |         59376 µs |
2021-11-03 10:20:06,571 # | atom |       or |    8 |         14063 µs |         18752 µs |         21876 µs |
2021-11-03 10:20:06,630 # | atom |       or |   16 |         12501 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:06,687 # | atom |       or |   32 |         10939 µs |         18751 µs |         20313 µs |
2021-11-03 10:20:06,802 # | atom |       or |   64 |         18751 µs |         23439 µs |         65626 µs |
2021-11-03 10:20:06,863 # | atom |      xor |    8 |         15626 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:06,934 # | atom |      xor |   16 |         12501 µs |         25001 µs |         26563 µs |
2021-11-03 10:20:06,996 # | atom |      xor |   32 |         10939 µs |         23438 µs |         20314 µs |
2021-11-03 10:20:07,128 # | atom |      xor |   64 |         26563 µs |         35939 µs |         62501 µs |
2021-11-03 10:20:07,189 # | atom |      and |    8 |         10938 µs |         18751 µs |         25002 µs |
2021-11-03 10:20:07,254 # | atom |      and |   16 |         15626 µs |         18752 µs |         23438 µs |
2021-11-03 10:20:07,321 # | atom |      and |   32 |         15626 µs |         18752 µs |         25001 µs |
2021-11-03 10:20:07,448 # | atom |      and |   64 |         28126 µs |         31251 µs |         60939 µs |
2021-11-03 10:20:07,505 # | atom |      set |    8 |         12501 µs |         17189 µs |         20313 µs |
2021-11-03 10:20:07,573 # | atom |      set |   16 |         17189 µs |         23438 µs |         20314 µs |
2021-11-03 10:20:07,633 # | atom |      set |   32 |         10939 µs |         21876 µs |         20313 µs |
2021-11-03 10:20:07,766 # | atom |      set |   64 |         28126 µs |         37502 µs |         60938 µs |
2021-11-03 10:20:07,823 # | atom |    clear |    8 |         10939 µs |         18751 µs |         20314 µs |
2021-11-03 10:20:07,891 # | atom |    clear |   16 |         18751 µs |         21876 µs |         20314 µs |
2021-11-03 10:20:07,954 # | atom |    clear |   32 |         10939 µs |         21876 µs |         23439 µs |
2021-11-03 10:20:08,079 # | atom |    clear |   64 |         26563 µs |         31252 µs |         60939 µs |
2021-11-03 10:20:08,157 # | semi |      add |    8 |         14063 µs |         35939 µs |         20314 µs |
2021-11-03 10:20:08,232 # | semi |      add |   16 |         12501 µs |         35939 µs |         20313 µs |
2021-11-03 10:20:08,319 # | semi |      add |   32 |         10938 µs |         43752 µs |         25001 µs |
2021-11-03 10:20:08,449 # | semi |      add |   64 |         25001 µs |         35939 µs |         62501 µs |
2021-11-03 10:20:08,523 # | semi |      sub |    8 |         12501 µs |         35939 µs |         18751 µs |
2021-11-03 10:20:08,597 # | semi |      sub |   16 |         12501 µs |         35939 µs |         18751 µs |
2021-11-03 10:20:08,685 # | semi |      sub |   32 |         15627 µs |         39063 µs |         26564 µs |
2021-11-03 10:20:08,792 # | semi |      sub |   64 |         17189 µs |         23439 µs |         59376 µs |
2021-11-03 10:20:08,867 # | semi |       or |    8 |         10939 µs |         37501 µs |         20314 µs |
2021-11-03 10:20:08,955 # | semi |       or |   16 |         12501 µs |         45314 µs |         23439 µs |
2021-11-03 10:20:09,034 # | semi |       or |   32 |         10938 µs |         40627 µs |         20313 µs |
2021-11-03 10:20:09,164 # | semi |       or |   64 |         26563 µs |         35938 µs |         60939 µs |
2021-11-03 10:20:09,244 # | semi |      xor |    8 |         12501 µs |         35939 µs |         25001 µs |
2021-11-03 10:20:09,328 # | semi |      xor |   16 |         14064 µs |         39064 µs |         23438 µs |
2021-11-03 10:20:09,397 # | semi |      xor |   32 |         10939 µs |         31251 µs |         20314 µs |
2021-11-03 10:20:09,522 # | semi |      xor |   64 |         25001 µs |         31251 µs |         62502 µs |
2021-11-03 10:20:09,597 # | semi |      and |    8 |         10939 µs |         35938 µs |         20314 µs |
2021-11-03 10:20:09,675 # | semi |      and |   16 |         10938 µs |         35938 µs |         25002 µs |
2021-11-03 10:20:09,751 # | semi |      and |   32 |         10939 µs |         37501 µs |         20313 µs |
2021-11-03 10:20:09,882 # | semi |      and |   64 |         28126 µs |         31251 µs |         65626 µs |
2021-11-03 10:20:09,890 # +------+----------+------+------------------+------------------+------------------+

```

</details>

There are some large differences between these two. Might be a cause for concern later, but I don't expect this PR to be the cause for this.

File: tests/bench_thread_yield_pingpong/main.c [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 #include "macros/units.h"
```
Comment (bergzand) [inline tests/bench_thread_yield_pingpong/main.c:23]: This test is still missing the includes for `CLOCK_CORECLOCK`

```suggestion
#include "periph_conf.h"
#include "timex.h"
#include "macros/units.h"
```

**Review by bergzand — CHANGES_REQUESTED** [review]

**Comment by fjmolinas** [comment]
@kaspar030 seems like its only missing some dependencies and board blacklisting

**Comment by fjmolinas** [comment]
Please squash!

**Comment by OlegHahm** [comment]
@kaspar030, ping!

**Comment by kaspar030** [comment]
I'm not going to work on this anymore.

**Pull request closed without merging**